### PR TITLE
httm: 0.23.3 -> 0.31.6

### PR DESCRIPTION
**Pull request by magnetophon**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/httm/default.nix [hunk 1]
```diff
@@ -6,29 +6,41 @@
...
 
   src = fetchFromGitHub {
     owner = "kimono-koans";
     repo = pname;
     rev = version;
-    hash = "sha256-yia7GEPemFVHzTkhrL7HejQsFO1zwpdUtq4DLdm4s2g=";
+    hash = "sha256-iASl4chx0vcS+SWK65URWqsGmucFUavKrDBnnaffLu8=";
   };
 
+  cargoHash = "";
```
Comment (Flakebi) [inline pkgs/tools/filesystems/httm/default.nix:18]: This shouldn’t be needed.

**Review by Flakebi — COMMENTED** [review]
Version is already at 0.38.1 now (which fixes using it with btrbk for me).
With 0.38.1, overwriting the `outputHashes` doesn’t seem to be required anymore.

File: pkgs/tools/filesystems/httm/default.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/tools/filesystems/httm/default.nix:18]: I forgot how to update this. :sweat_smile: 
No matter if with or without that line, I get:
```
 nix profile install -f $NIXPKGS httm
error: build of '/nix/store/s4vqvaw8rsfpim196xg0sg5yxv2qp9lr-httm-0.38.1.drv' on 'ssh://nixBuild@builder' failed: builder for '/nix/store/s4vqvaw8rsfpim196xg0sg5yxv2qp9lr-httm-0.38.1.drv' failed with exit code 1;
       last 10 log lines:
       > ERROR: cargoHash or cargoSha256 is out of date
       >
       > Cargo.lock is not the same in /build/cargo-vendor-dir
       >
       > To fix the issue:
       > 1. Set cargoHash/cargoSha256 to an empty string: `cargoHash = "";`
       > 2. Build the derivation and wait for it to fail with a hash mismatch
       > 3. Copy the "got: sha256-..." value back into the cargoHash field
       >    You should have: cargoHash = "sha256-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX=";
       >
       For full logs, run 'nix-store -l /nix/store/s4vqvaw8rsfpim196xg0sg5yxv2qp9lr-httm-0.38.1.drv'.
error: builder for '/nix/store/s4vqvaw8rsfpim196xg0sg5yxv2qp9lr-httm-0.38.1.drv' failed with exit code 1;
       last 10 log lines:
       > ERROR: cargoHash or cargoSha256 is out of date
       >
       > Cargo.lock is not the same in /build/cargo-vendor-dir
       >
       > To fix the issue:
       > 1. Set cargoHash/cargoSha256 to an empty string: `cargoHash = "";`
       > 2. Build the derivation and wait for it to fail with a hash mismatch
       > 3. Copy the "got: sha256-..." value back into the cargoHash field
       >    You should have: cargoHash = "sha256-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX=";
       >
       For full logs, run 'nix log /nix/store/s4vqvaw8rsfpim196xg0sg5yxv2qp9lr-httm-0.38.1.drv'.
       ```

Comment (Flakebi) [reply pkgs/tools/filesystems/httm/default.nix:18]: Removing this line and updating the `Cargo.lock` in nixpkgs worked for me

Comment (magnetophon) [reply pkgs/tools/filesystems/httm/default.nix:18]: Could you open a new PR then?

Comment (Flakebi) [reply pkgs/tools/filesystems/httm/default.nix:18]: Here you go: #306783

**Pull request closed without merging**